### PR TITLE
Adding stlearn recipe

### PR DESCRIPTION
**Pull request by duypham2108**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by duypham2108** [comment]
@conda-forge/help-python The recipe has passed all checks and ready for your review. Thank you

**Comment by duypham2108** [comment]
@conda-forge/help-python The recipe has passed all checks and ready for your review. Thank you

File: recipes/stlearn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - tensorflow
+    - leidenalg
+    - louvain
+    - xlrd
+
+
+test:
+  imports:
+    - stlearn
+  requires:
```
Comment (chrisburr) [inline recipes/stlearn/meta.yaml:40]: Are you missing a pip check?
```suggestion
  commands:
    - pip check
  requires:
```

File: recipes/stlearn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - pip
+    - python >=3.7
+  run:
+    - bokeh ==2.4.2
+    - numba
+    - numpy ==1.21.5
+    - pillow ==9.0.1
+    - python >=3.7
+    - scanpy ==1.8.2
+    - scikit-image ==0.19.2
```
Comment (chrisburr) [inline recipes/stlearn/meta.yaml:30]: Pinning exactly like this almost always a bad idea that causes trouble. If you have two packages doing this they become impossible to use together.

For a longer explanation of the issues see: https://iscinumpy.dev/post/bound-version-constraints/

Comment (duypham2108) [reply recipes/stlearn/meta.yaml]: Thank you for your feedback. I have already fixed it. Please check it again

**Comment by duypham2108** [comment]
@conda-forge/help-python The recipe has fixed and passed all checks and ready for your review again. Thank you

**Comment by ngam** [comment]
> @conda-forge/help-python

because you're a first-time contributor, this didn't work, so I will call them for you: @conda-forge/help-python

File: recipes/stlearn/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - scanpy >=1.8.2
+    - scikit-image >=0.19.2
+    - tensorflow
+    - xlrd <2.0
+
+test:
+  imports:
+    - stlearn
+  commands:
+    - pip check
```
Comment (BastianZim) [inline recipes/stlearn/meta.yaml:39]:
```suggestion
    - pip check
    - stlearn --help
```
Just to test the entry points

File: recipes/stlearn/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - bokeh >=2.4.2
+    - leidenalg
+    - louvain
+    - numpy >=1.18,<1.22
+    - pillow >=9.0.1
+    - python >=3.7
+    - scanpy >=1.8.2
+    - scikit-image >=0.19.2
+    - tensorflow
+    - xlrd <2.0
```
Comment (BastianZim) [inline recipes/stlearn/meta.yaml:33]: I can't find this one upstream, are you sure about this?

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just two small things, then this is ready to go.

File: recipes/stlearn/meta.yaml (hunk 4 above)
Comment (duypham2108) [reply recipes/stlearn/meta.yaml:34]: Sorry, I'm not sure I understand what you mean. Anyways, I have updated the new version of the package

Comment (BastianZim) [reply recipes/stlearn/meta.yaml:34]: xlrd was not listed as a dependency upstream .

Comment (duypham2108) [reply recipes/stlearn/meta.yaml]: I fixed it, thanks

**Review by BastianZim — APPROVED** [review]

**Pull request merged**